mobile/src/components/courses/courses.tsx

```diff
@@ -49,26 +57,58 @@ const Courses: FC = (): ReactElement => {
   };
 
   const handleSearch = (search: string): void => {
-    dispatch(coursesActions.getCourses({ title: search, categoryKey: '' }));
+    if (!courseCategory) {
+      dispatch(coursesActions.getCourses({ title: search, categoryKey: '' }));
+    } else {
+      dispatch(
+        coursesActions.getCourses({
+          title: search,
+          categoryKey: courseCategory.key,
+        }),
+      );
+    }
   };
 
+  useEffect(() => {
+    if (courseCategory) {
+      dispatch(
+        coursesActions.getCourses({
+          title: '',
+          categoryKey: courseCategory.key,
+        }),
+      );
+    } else {
+      dispatch(coursesActions.getCourses({ title: '', categoryKey: '' }));
+    }
+  }, [courseCategory]);
+
+  useEffect(() => {
+    dispatch(categoryActions.clearCategory());
+    dispatch(categoryActions.getCategories());
```

Comment:
Why do you need clearCategory here?

Reply:
To keep category null every time we leave the courses screen and go back. For example, we select the category, then go to anouther screen and return to courses. Category list will reset its values but there will be the category in store and it will be the situation, when we won't have selected category but but the courses list will be fetched by saved in store category
I hope I didn't confuse you too much)

Reply:
You use useEffect instead of useFocusEffect here, so it will be executed, only when component mounts, not when we go to this screen